app/models/course/assessment/question.rb

```diff
@@ -5,4 +5,17 @@ class Course::Assessment::Question < ActiveRecord::Base
   has_and_belongs_to_many :tags
 
   delegate :to_partial_path, to: :actable
+
+  # Attempts the given question in the submission. This builds a new answer for the current
+  # question.
+  #
+  # @param [Course::Assessment::Submission] submission The submission which the answer should
+  #   belong to.
+  # @return [Course::Assessment::Answer] The answer corresponding to the question. It is required
+  #   that the {Course::Assessment::Answer#question} property be the same as +self+. The result
+  #   should not be persisted.
+  def attempt(submission)
```

Comment:
I don't think there's a way to implement the attempt method to the user, because it's not the user that does an attempt, because the student cab attempt the same assessment multiple times, like in trainings.